fuel-core/src/database/message.rs

```diff
@@ -13,12 +19,31 @@ impl Storage<MessageId, DaMessage> for Database {
         key: &MessageId,
         value: &DaMessage,
     ) -> Result<Option<DaMessage>, KvStoreError> {
+        // insert secondary record by owner
+        Database::insert(
```

Comment:
It seems like we should do the primary insert of the record before the updating the index. Just in case something tries to query the index and the underlying data isn't persisted yet.